spec/requests/app_spec.rb

```diff
@@ -46,6 +48,35 @@
     end
   end
 
+  describe 'POST /user/residents' do
+    context 'when the file is valid' do
+      it 'creates a resident for each name in the file' do
+        post '/user/residents', { file: pdf_file }, headers
+
+        expect(last_response.status).to eq(200)
+        expect(Resident.count).to eq(23)
```

Comment:
⛏️ Usually we'd see this as `expect { post ... }.to change(Resident, :count).by(23)`. We're using DatabaseCleaner so I don't think we could actually get a false positive in practice, but it's a little bit more "correct" to do it this way.